### PR TITLE
Math magicians: tests

### PR DESCRIPTION
**Pull request by AlselwiAisha**
**Description:**

- Add tests to the Math Magicians app using [Jest](https://jestjs.io/) and [React Testing Library](https://testing-library.com/docs/react-testing-library/intro).
- Write unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all React components in **components.test.js** file
- Use React Testing Library to simulate user interaction.
- Run full tests suite using Jest. All tests pass.

### CODE REVIEW

File: src/Test/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  },
+];
+
+for (let i = 0; i < objParamAc.length; i += 1) {
+  test('calculate function returns object based on parameter', () => {
+    expect(calculate(
+      objParamAc[i].identifier, objParamAc[i].identifier,
+    )).toEqual(objParamAc[i].returnObj);
+  });
+}
```
Comment (julie-ify) [inline src/Test/calculate.test.js]: - [ ] Kindly add more tests to cover all the scenarios in the calculate.js file

File: src/Test/Components.test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
```
Comment (julie-ify) [inline src/Test/Components.test.js:3]: - [ ] Using the react testing library, kindly add tests to simulate user interactions, like firing events and so on.

  You can visit [this tutorial](https://www.robinwieruch.de/react-testing-library/) for ideas.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @AlselwiAisha and @Yonasle,

Good job so far!

## Highlights!
- Good code organization and tests ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Bantealem — APPROVED** [review]
# Status: Approved
Hi @AlselwiAisha and @Yonasle 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [x] No Linters Error
- [x] Professional Readme
- [x] Correct Git flow
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**